### PR TITLE
Give DurableObjectStorage reference to primary

### PR DESCRIPTION
**Pull request by MellowYarker**

### CODE REVIEW

**Comment by justin-mp** [comment]
> @justin-mp alright if I consolidate to 1 commit and mark you as a co-author?

Fine with me.

File: src/workerd/io/io-channels.h [hunk 1]
```diff
@@ -4,6 +4,9 @@
 
 #pragma once
 
+#include "workerd/io/actor-cache.h"
+#include "workerd/io/io-own.h"
```
Comment (jasnell) [inline src/workerd/io/io-channels.h:8]: Nit: are these includes actually needed in io-channels.h or is this just to make the decls available in the other files changed in this PR? If the latter, I'd prefer if the includes were added where they are actually used.

**Review by jasnell — APPROVED** [review]

**Review by jasnell — APPROVED** [review]

**Comment by jasnell** [comment]
A PR so nice I approved it thrice. 

image

Really not sure what happened here, Github.

File: src/workerd/api/actor-state.h [hunk 2]
```diff
@@ -177,6 +179,16 @@ class DurableObjectStorage: public jsg::Object, public DurableObjectStorageOpera
       : cache(kj::mv(cache)),
         enableSql(enableSql) {}
 
+  // This constructor is only used when we're setting up the `DurableObjectStorage` for a replica
+  // Durable Object instance. Replica's need to retain a reference to their primary so they can
```
Comment (justin-mp) [inline src/workerd/api/actor-state.h:183]: nit: Replicas (plural, not possessive)

**Review by justin-mp — COMMENTED** [review]
LGTM, but please squash the 2 commits and clean up the files in src/workerd/io since I don't think we need any of those changes.

**Review by justin-mp — APPROVED** [review]

File: src/workerd/io/io-channels.h (hunk 1 above)
Comment (MellowYarker) [reply src/workerd/io/io-channels.h]: Had initially needed these includes, but took the PR in another direction. Thanks for catching!

**Comment by kentonv** [comment]
FYI, this build was merged with CI failing. It was an "optional" CI build, but the failure was in fact caused by this change and so should have been considered blocking.

**Pull request merged**